src/models/magitLogCommit.ts

```diff
@@ -0,0 +1,35 @@
+import { Commit } from '../typings/git';
+
+interface IMagitLogCommit {
+  graph: string[] | undefined;
+  hash: string;
+  refs: string | undefined;
+  author: string;
+  time: Date;
+  message: string;
+}
+
+export class MagitLogCommit implements Commit, IMagitLogCommit {
```

Comment:
There are no classes in `models/` this should just be an interface. Don't want any logic in here.
You can use the `extends` keyword to extend other interfaces.

Note: As for MagitError being a class; this is because its used in the `exception` style so I wanted a constructor. This the exception to the rule.

Reply:
Got you. I moved the logic out of this file. Let me know if that's not you want.

Reply:
https://github.com/stevenguh/edamagit/pull/1